### DISCUSSION TITLE
Routing in the same level to load files (`$filename[.]$ext`)

### DISCUSSION THREAD

**Discussion by jmgirao** [remix-run/remix#5879] [Q&A] [answered]
Hi team!

I have one question about routing in remix. I need to load the static files (txt, html, xml, ...), these files are provided by external API.

my routes are:
/$id
/-- index
$id.ts
$filename[.]$ext.ts

When I access `my-file.txt`, the route should be `$filename[.]$ext` but the loaded route is `$id`. 

if the file route was $filename[.txt], it's work, but my goal the file extension was dynamic.

any suggestion? 
thanks

**Comment by SeraphicRav** [comment, marked as answer]
Hello !

Sorry, not a member of the team but I believe that it is not possible in router v1 (I do not not v2).

The router use segments so basically $id.ts and $filename[.]$ext.ts are the same path segment and thus can only be referred as 1 variable.

I think the solution would be to check in $id.tsx what is the value of id:
- if it is a number do processing of id 
- if file name, do the file logic depending
